apis/v1alpha1/httproute_types.go

```diff
@@ -58,6 +58,16 @@ type HTTPRouteHost struct {
 	// +optional
 	Hostnames []string `json:"hostnames,omitempty"`
 
+	// TLS defines the TLS certificate to use for Hostnames defined in this
+	// Route. The certificate defined here takes effect only if the
+	// AllowOverrideViaRoute field is set to true in the associated Gateway
+	// resource.
+	//
+	// Support: Core
+	//
+	// +optional
+	TLS *RouteTLSConfig `json:"tls" protobuf:"bytes,2,opt,name=tls"`
```

Comment:
Why not use `TLSConfig`? I know that `Mode` is not applicable here, but can't that be addressed in docs?

Reply:
One of the bits of feedback from Tim was to avoid recycling types quite so much, especially if not all fields applied in all cases. I think it's useful to ensure our TLSConfig types are consistent but I'd rather not have extra fields in the API that can't actually be used.

Reply:
I initially did that but decided against for several reasons:
- The generated CRD will have the fields and it can and probably will confuse users if the fields are present.
- Implementers that come in later could misinterpret things.
- We might add additional fields at the Gateway-level which will automatically appear here. I think it would be better for us to make a conscious decisions to introduce fields at this level instead.
- Based on the previous discussions, it seems that providing ability to configure certificate is the most asked for requirement. It is unlikely that we will want to expand TLS at this level much more.

Reply:
Got it, thanks for the additional input.